environment/src/main/java/com/sequenceiq/environment/environment/repository/EnvironmentRepository.java

```diff
@@ -73,4 +73,8 @@
             + "WHERE pe.id = :parentEnvironmentId AND e.accountId = :accountId AND e.archived = false")
     List<Environment> findAllByAccountIdAndParentEnvIdAndArchivedIsFalse(@Param("accountId") String accountId,
             @Param("parentEnvironmentId") Long parentEnvironmentId);
+
+    @CheckPermission(action = ResourceAction.READ)
+    @Query("SELECT e FROM Environment e WHERE e.archived = false and e.status in (:statuses)")
+    List<Environment> findAllRunningAndStatusIn(@Param("statuses") Collection<EnvironmentStatus> statuses);
```

Comment:
I guess the view is enough right ?

Reply:
you are right, I'm going to replace

Reply:
No, I cannot replace. It is run only once at start only. This code is used in another place so I should convert there the Environment to EnvironmentView. But I can optimize the fetch part like: `e.id, e.name, e.resourceCrn`